src/test/java/codeu/controller/NotificationServletTest.java

```diff
@@ -0,0 +1,42 @@
+package codeu.controller;
+
+import java.io.IOException;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class NotificationServletTest {
+
+  private HttpServletRequest mockRequest;
+  private HttpSession mockSession;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private NotificationServlet notificationServlet;
+
+  @Before
+  public void setup() {
+    notificationServlet = new NotificationServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/notifications.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
```

Comment:
What happens when the user is not logged in? "/notifications" still shows the page in the prototype version. This is okay for now I guess, but in the final stage we would want to show some server error that says "Not logged in" (and test that behavior too).

Add a TODO for this here, before merging

Reply:
^^ I second this